.github/workflows/ci.yml

```diff
@@ -0,0 +1,237 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - '*.x'
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - main
+  schedule:
+    # Weekly Monday 9AM build
+    # * is a special character in YAML so you have to quote this string
+    - cron: '0 9 * * 1'
+
+env:
+  CRDS_SERVER_URL: "https://roman-crds-test.stsci.edu"
+  CRDS_PATH: "/tmp/crds_cache"
+  CRDS_CLIENT_RETRY_COUNT: 3
+  CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
+
+jobs:
+  style:
+    name: Code style checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: 3.9
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: style-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install pyproject-flake8
+      - run: pip freeze
+      - run: pflake8
```

Comment:
Why not use `tox` environments?

Reply:
https://github.com/spacetelescope/romanisim/pull/14#discussion_r1006877851

Reply:
In general, IMOP we should be providing `tox` envs for the CI to run, rather than running a bunch of commands. This is because `tox` builds an exactly reproducible python environment for testing and then runs an identical set of commands.

Reply:
That is true, though is does add another level of complexity it does allow the user to reproduce the tests locally for their Python version

Reply:
I leave this to Zach's ticket.

Reply:
I'll make a separate PR to implement tox